services/ui-src/src/utils/autosave/autosave.ts

```diff
@@ -26,35 +25,50 @@ interface Props {
     userName: string | undefined;
     state: string | undefined;
   };
+  fieldType?: string;
 }
 
 export const autosaveFieldData = async ({
   form,
   fields,
   report,
   user,
+  fieldType,
 }: Props) => {
   const { id, updateReport } = report;
   const { userName, state } = user;
+  let onlyChangedFields = [];
+  switch (fieldType) {
+    case "dynamic":
+      onlyChangedFields = fields.filter((field: FieldInfo) => {
+        const checkedValues = field.displayValues?.filter((el) => el.name);
+        return (
+          checkedValues?.length !== 0 &&
+          field.displayValues !== field?.hydrationValue
+        );
+      });
+      break;
```

Comment:
Thought: If even one of the entity names has changed, we need to make the database call, in which case we might as well update all the entity names, so there's probably no need to go through each one individually. The top-level check of value vs. hydration value should be enough. 

We also don't need to pass through a separate `displayValues` prop here -- we can just pass it as the `value` prop that all fields use. That way we can just use the exact same code for dynamicFields that we use for everything else, _and_ avoid having to pass through any information about the field type.